src/main/rx/rx.c

```diff
@@ -746,14 +758,17 @@ uint8_t getRssiPercent(void)
 }
 
 #ifdef USE_RX_LINK_QUALITY_INFO
-uint8_t rxGetLinkQuality(void)
+uint16_t rxGetLinkQuality(void)
 {
     return linkQuality;
 }
 
-uint8_t rxGetLinkQualityPercent(void)
+uint16_t rxGetLinkQualityPercent(void)
 {
-    return scaleRange(rxGetLinkQuality(), 0, LINK_QUALITY_MAX_VALUE, 0, 100);
+    if (!linkQuality) {
```

Comment:
Not sure why you are adding this special case for 0 here - this should just be part of returning the result.

Reply:
failed my unit test. if i remember correctly .. ill check logic again ... do to rx loss sets it to 0.  but i do have a habit of rechecking items in code . for sanity .

Reply:
ok it just drops a few .. cycles of the cpu  No  need to scale 0 its 0 ...

Reply:
i think it should be done .  will remove if you like .

Reply:
@mikeller can i keep the test not to do extra cpu cycles ?

Reply:
That's what is called 'premature optimisation'. We are using aggressive compiler optimisation settings for the firmware, and the compiler is usually quite good at optimising things like this, so I doubt that this will give us any improvement. If the unit test failed then it's more likely that something was not right, and adding a workaround to make the test pass is definitely not the right solution, the problem should be identified and fixed.

Reply:
@mikeller 
ok that was unknown to me . i will remove and recheck.
please mark PR **testing needed** please.

Reply:
@mikeller removed  'premature optimization' ...  no issue with unit test